src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -216,6 +224,100 @@ export class QuickBar extends LitElement {
     }));
   }
 
+  private _generateServerControlCommandItems(): QuickBarItem[] {
+    const serverActions = ["restart", "stop"];
+
+    return serverActions.map((action) =>
+      this._generateConfirmationCommandItem(
+        {
+          text: this.hass.localize(
+            "ui.dialogs.quick-bar.commands.server_control.perform_action",
+            "action",
+            this.hass.localize(
+              `ui.dialogs.quick-bar.commands.server_control.${action}`
+            )
+          ),
+          icon: "hass:robot",
+          action: () => this.hass.callService("homeassistant", action),
+        },
+        this.hass.localize("ui.dialogs.generic.default_confirmation_title")
+      )
+    );
+  }
+
+  private _generateConfirmationCommandItem(
+    item: QuickBarItem,
+    confirmText: ConfirmationDialogParams["confirmText"]
+  ): QuickBarItem {
+    return {
+      ...item,
+      action: () =>
+        showConfirmationDialog(this, {
+          confirmText,
+          confirm: item.action,
+        }),
+    };
+  }
+
+  private getNavigationInfo(sectionKey: string, component: string) {
+    const panel = configSections[sectionKey].find(
+      (section) => section.component === component
+    );
+
+    if (!panel || !panel.translationKey) {
+      throw Error("Navigation section or its translation not found");
+    }
+
+    const text = this.hass.localize(
+      "ui.dialogs.quick-bar.commands.navigation.navigate_to",
+      "panel",
+      this.hass.localize(
+        `ui.dialogs.quick-bar.commands.navigation.${component}`
+      )
+    );
+
+    return { ...panel, text };
+  }
+
+  private _generateNavigationCommandItems(): QuickBarItem[] {
+    const panels = [
```

Comment:
We should use `this.hass.panels`

Reply:
Hmm. It doesn't look like `this.hass.panels` surfaces the configSection key I need for getting things like translationKey and full url path from `configSections`.

Here's what I'm getting in hass object:
![image](https://user-images.githubusercontent.com/1480827/96766570-244a4e80-1390-11eb-8b83-669e95a5fe29.png)

But I can't use that information to get to the data I need in the `configSections` object:
https://github.com/home-assistant/frontend/blob/611202c905271a6fe746c7d86eca7ef956abca9d/src/panels/config/ha-panel-config.ts#L45-L182

Reply:
I could maybe add the data I need to that object? Also, why the heck does `url_path` not actually give the full url path?

Reply:
Done